core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java

```diff
@@ -97,8 +97,13 @@ public void authenticate(String connectionUser, String impersonationUser)
     Set<String> allowedGroups = mImpersonationGroups.get(connectionUser);
 
     if (allowedUsers == null && allowedGroups == null) {
-      throw new AuthenticationException(ExceptionMessage.IMPERSONATION_NOT_CONFIGURED
-          .getMessage(connectionUser, impersonationUser));
+      throw new AuthenticationException(String.format(
+          "Failed to authenticate client user=\"%s\" connecting to Alluxio master and "
```

Comment:
I think technically this exception can happen on any alluxio server, master or worker.

Reply:
oh? do we also authenticate clients on workers?